sdktests/server_side_big_segments.go

```diff
@@ -183,18 +213,56 @@ func doBigSegmentsMembershipCachingTests(t *ldtest.T) {
 		assert.Equal(t, []string{expectedUserHash1}, bigSegmentStore.GetMembershipQueries())
 	})
 
+	t.Run("queries may be done for multiple context kinds", func(t *ldtest.T) {
+		// Here we provide a multi-kind context where there is at least one rule checking each kind,
+		// with a different key for each kind.
```

Comment:
These two tests are new, and cover the biggest change to the big segments evaluation logic: the evaluation engine may now do more than one big segments query, if you have a multi-kind context and it has to evaluate segments that use different kinds (and the keys for those kinds are not the same).